149_add-shift-data-preparation_22_issue149

MiguelQuintana1 opened the pull request:
This PR will solve issue #149 

https://github.com/user-attachments/assets/1dbeb74e-18c3-4a84-9b57-39107b17b688


## PR Description: Retool AddShift Component for Dummy Shift Data Generation

This pull request refactors the `AddShift` component to streamline the process of generating dummy shift data, focusing specifically on the **Date**, **Start Time**, and **End Time** fields as requested. The previous version included fields for Role, Location, and Notes, which have been removed to meet the updated requirements.

**Key Changes:**

* **Simplified Component:** The component now presents a cleaner interface with input mechanisms only for Date, Start Time, and End Time.
* **Date Handling:**
    * A `DatePicker` is implemented using `@react-native-community/datetimepicker` (or a web `date` input on web platforms) to allow users to select a specific date.
    * A `formatDate` function ensures the selected date is formatted as `YYYY-MM-DD`.
* **Time Handling:**
    * Time pickers are implemented using `@react-native-community/datetimepicker` to select the start and end times.
    * A `formatTime` function formats the selected times as `HH:MM:SS+00`.
* **Dummy Data Generation:**
    * When the "Add Shift" button is pressed, the `handleAddShift` function formats the selected date, start time, and end time into the expected output format: `{ 'YYYY-MM-DD HH:MM:SS+00', 'YYYY-MM-DD HH:MM:SS+00' }`.
    * This formatted data is currently logged to the console. In a real application, this function would likely be adapted to send this data to an API or update local state.
    * Improved user feedback on button press with a temporary "Adding..." state and subtle visual changes.
* **Styling Enhancements:** The component's visual presentation has been updated with a more modern and consistent style, including:
    * A defined color palette for better theming.
    * Subtle shadows for depth.
    * Styled input fields and buttons for improved aesthetics and user interaction.
    * A toned-down primary button color for a less "loud" appearance.
* **Type Safety:** Addressed TypeScript linting issues by explicitly typing the `event` parameters in the date and time picker change handlers and using `type` imports where appropriate.
* **Cross-Platform Considerations:** While aiming for cross-platform compatibility with React Native, the PR acknowledges potential minor differences in native UI elements (like date/time pickers) between iOS and Android, recommending thorough testing on both platforms.

**Purpose of this Change:**

This refactored component provides a focused tool for generating dummy shift data in the specified format, which can be valuable for testing, prototyping, or populating development environments. By removing the unnecessary fields and enhancing the UI and data formatting, this PR delivers a more efficient and user-friendly experience for this specific task.

MiguelQuintana1 committed:
update packages

update packages

MiguelQuintana1 committed:
i modified the AddShift component preparing for data integration, as well as i modified the style for a better visual experience

cshimm commented on package.json:
```diff
@@ -31,7 +31,7 @@
     "@expo/vector-icons": "^14.0.4",
     "@react-native-async-storage/async-storage": "^2.1.0",
-    "@react-native-community/datetimepicker": "8.2.0",
+    "@react-native-community/datetimepicker": "^8.3.0",
```
There is an issue with version `"^8.3.0"` for the React Native Community datetimepicker on Android. Downgrading to version `"8.2.0"` resolved the issue.

![image](https://github.com/user-attachments/assets/3dc5594d-02b5-4e65-b6cb-02deaa8ef416)

cshimm reviewed (dismissed):
The changes work as expected. I was able to select a date, start time, end time and log the shift to the console with the add shift button. I was though having issues with the React Native Community DateTimePicker component, specifically with the version It was updated to. downgrading to the version below resolved the issue. I'll be sure to approve once the package version issue is resolved.

For future iterations we can implement guards to ensure an even stronger UX for this component. This would include things like
- Disallowing shifts to be created where the end time is before the start time
- Limiting minute values to quarter hour values (XX:00, XX:15, XX:30, etc)
- Excluding seconds value from time picker
- Formatting date in a more human readable way (ie Monday, January 1 2025 or something similar)

nwm516 committed:
Change datetimepicker version

Changed datetimepicker dependency version to 8.2.0 to work with Android due to issues from version 8.3.0

IsaacJrTypes commented:
I agree with Cameron's suggestions, while I personally prefer military time because its easier to tell time when traveling countries, regular folks don't lol. You can use the date object's instance methods to help you convert military time to standard time and also extract the UTC value. Check out the links below to help you out. Overall, good job!

- https://github.com/SeattleColleges/shift-app-expo/pull/150/commits/1f74134c6ad96132d421aab6bebe686cf6aecc33
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date

MiguelQuintana1 committed:
new functionality suggested by Cameron

MiguelQuintana1 commented:
I modified the format of the data; now it looks like this:

Shift added: { 'Friday, May 16, 2025 23:00', 'Saturday, May 17, 2025 05:00' }

Also, now if the date/time is less than the starting date/time, it will show an error message. 

Another modification is the time you pick; it will round up to the closest quarter. If it is 5:11pm, it will round up to 5:15pm.

Thank you guys for the suggestions!!!.

MiguelQuintana1 commented:
https://github.com/user-attachments/assets/ccc1220c-658c-4239-9295-8a1165f51f88

MiguelQuintana1 committed:
update CLi

same

MiguelQuintana1 commented:
Also we i re-worked the ShiftApp component to be Mobile native focused by:

1. **Removed Web-Specific Logic**:
   - Eliminated any web-specific handling or checks, ensuring the component is optimized for mobile platforms.

2. **Focused on Native Components**:
   - Used React Native's `Modal` and `DateTimePicker` directly without conditional logic for web compatibility.

3. **Simplified Modal Handling**:
   - Unified the modal logic for both date and time pickers, making it cleaner and easier to maintain.

4. **Optimized Styling**:
   - Ensured the styles are responsive and suitable for smaller mobile screens, using padding, alignment, and flexible layouts.

5. **Improved Code Readability**:
   - Refactored the code to make it more concise and focused on mobile-native behavior, removing unnecessary complexity.

These changes ensure the `AddShift` component is fully optimized for mobile devices, providing a better user experience and cleaner codebase.

MiguelQuintana1 committed:
reworked component Mobile Focused

nwm516 committed:
Merge branch 'main' into 149-add-shift-data-preparation

nwm516 committed:
Create package-lock.json

nwm516 committed:
Update AddShift.tsx
